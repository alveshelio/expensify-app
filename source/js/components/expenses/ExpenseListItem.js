import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

import { removeExpense } from '../../actions/expensesActions';

const ExpenseListItem = ({
  id,
  description,
  amount,
  note,
  createdAt,
  sortBy,
  text,
  dispatch,
  history,
}) => {
  return (
    <div className='expense-list-item'>
      <span>Description: {description}</span>
      {note && <span>Note: {note}</span>}
      <span>Amount: {amount}</span>
      {createdAt && <span>Created On: {createdAt.format('MMM Do, YYYY')}</span>}
      <button onClick={() => dispatch(removeExpense(id))}>Remove</button>
      <Link to={`/edit/${id}`}>Edit</Link>
      <hr />
    </div>
  );
};

ExpenseListItem.propTypes = {
  id: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  amount: PropTypes.string.isRequired,
  note: PropTypes.string,
  createdAt: PropTypes.shape({

  }),
  sortBy: PropTypes.string,
  text: PropTypes.string,
};

// const mapStateToProps = state => ({
//   expenses: state.expenses,
// });

export default connect()(ExpenseListItem);
