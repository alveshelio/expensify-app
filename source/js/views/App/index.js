import React from 'react';

const App = () => (
  <div>
    <h1>App is running on port 3000</h1>
  </div>
);

export default App;
