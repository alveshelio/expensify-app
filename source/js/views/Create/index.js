import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import CreateExpenseForm from '../../components/expenses/forms/ExpenseForm';
import { addExpense } from '../../actions/expensesActions';

const CreateExpensePage = ({ dispatch, history }) => (
  <div>
    <h1>Create</h1>
    <CreateExpenseForm
      onSubmit={(expense) => {
        dispatch(addExpense(expense));
        history.push('/');
      }}
    />
  </div>
);

CreateExpensePage.propTypes = {
  dispatch: PropTypes.func.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
};

export default connect()(CreateExpensePage);
