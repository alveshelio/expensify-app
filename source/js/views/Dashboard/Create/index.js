import React from 'react';

const Create = () => (
  <div>
    <h1>Create</h1>
    <p>Dashboard Create expense</p>
  </div>
);

export default Create;
