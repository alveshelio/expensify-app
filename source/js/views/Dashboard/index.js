import React from 'react';
import DashboardRoutes from '../../routes/DashboardRoutes';

const Dashboard = () => (
  <div>
    <h1>Dashboard</h1>
    <DashboardRoutes />
  </div>
);

export default Dashboard;
