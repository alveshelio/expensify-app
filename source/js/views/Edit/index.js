import React from 'react';

const Edit = ({ match }) => (
  <div>
    <h1>Edit</h1>
    <p>Editing expense ID: {match.params.id}</p>
  </div>
);

export default Edit;
