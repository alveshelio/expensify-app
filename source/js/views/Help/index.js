import React from 'react';

const Help = (props) => {
  console.log('props', props);
  return (
    <div>
      <h1>Help Page</h1>
    </div>
  );
};

export default Help;
