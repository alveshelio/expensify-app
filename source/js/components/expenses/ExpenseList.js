import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../../selectors/expensesSelectors';
import { removeExpense } from '../../actions/expensesActions';

export const ExpenseList = ({ expenses, filters, dispatch }) => {
  return (
    <div>
      <h1>ExpenseList</h1>
      {
        expenses.length > 0 ?
          expenses.map(expense =>
            (<ExpenseListItem
              key={expense.id}
              {...filters}
              {...expense}
              onRemoveHandle={id => dispatch(removeExpense(id))}
            />)) :
          <p>You don&apos;t have any expenses yet</p>
      }
    </div>
  );
};

ExpenseList.propTypes = {
  expenses: PropTypes.arrayOf(PropTypes.shape({
    description: PropTypes.string.isRequired,
    amount: PropTypes.number.isRequired,
    note: PropTypes.string,
    createdAt: PropTypes.number.isRequired,
  }).isRequired).isRequired,
  filters: PropTypes.shape({
    text: PropTypes.string,
    sortBy: PropTypes.string,
    startDate: PropTypes.number,
    endDate: PropTypes.number,
  }),
  dispatch: PropTypes.func,
};

const mapStateToProps = state => ({
  expenses: selectExpenses(state.expenses, state.filters),
});

export default connect(mapStateToProps)(ExpenseList);
