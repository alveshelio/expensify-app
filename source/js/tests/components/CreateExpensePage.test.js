/* eslint no-undef: "off" */

import React from 'react';
import { shallow } from 'enzyme';
import { CreateExpensePage } from '../../views/Create';

import expenses from '../../fixtures/expenses';

test('should render Create Expense Page correctly', () => {
  const onSubmitSpy = jest.fn();
  const history = { push: jest.fn() };
  const wrapper = shallow(<CreateExpensePage onSubmit={onSubmitSpy} history={history} />);
  expect(wrapper).toMatchSnapshot();
});

test('should handle onSubmit', () => {
  const onSubmitSpy = jest.fn();
  const history = { push: jest.fn() };
  const wrapper = shallow(<CreateExpensePage onSubmit={onSubmitSpy} history={history} />);
  wrapper.find('ExpenseForm').prop('onSubmit')(expenses[0]);
  // expect(history).toHaveBeenLastCalledWith('/');
  expect(onSubmitSpy).toHaveBeenLastCalledWith({ expense: expenses[0] });
});
