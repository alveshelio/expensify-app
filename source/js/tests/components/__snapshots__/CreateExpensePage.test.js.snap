// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Create Expense Page correctly 1`] = `
<div>
  <h1>
    Create
  </h1>
  <ExpenseForm
    onSubmit={[Function]}
  />
</div>
`;
